script_editor/app/views/plays/show.html.erb

```diff
@@ -364,18 +366,17 @@
             }
         }
 
-        function mouseUp(e) {
+        function mouseUp({ srcElement }) {
             if (lowIndex == highIndex) {
-                e.srcElement.dataset.cut == "true" ? uncut(e.srcElement) : cut(e.srcElement)
+                srcElement.dataset.cut == "true" ? uncut(srcElement) : cut(srcElement)
             } else {
-                Array.from(e.srcElement.parentNode.children).slice(lowIndex, highIndex + 1).forEach(
+                Array.from(parent.children).slice(lowIndex, highIndex + 1).forEach(
```

Comment:
Why does defining srcElement.parentNode on the mouse down event, then reseting it for every highlight fix our problem? shouldn't srcElement.parentNode on mouse down be the same as on mouse up?

Reply:
If you mouseUp over something that's not a word or punc or something, e.srcElement.parent won't be the script element, it'll be something else. The indexes we save refer to the index of the children of the script element, so the parent variable should always be the correct script element.